metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -346,7 +371,7 @@ def emit_custom_sql_datasources(self) -> Iterable[MetadataWorkUnit]:
                 custom_sql_connection.get("nodes", [])
             )
             for csql in unique_custom_sql:
-                csql_id: str = csql.get("id", "")
+                csql_id: str = csql["id"]
```

Comment:
why this change? what if there is no `id`?

Reply:
`id` is mandatory field in `CustomSQLTable` graphql type https://help.tableau.com/current/api/metadata_api/en-us/reference/customsqltable.doc.html